source/material_model/rheology/visco_plastic.cc

```diff
@@ -601,11 +621,23 @@ namespace aspect
         // Reference and minimum/maximum values
         min_strain_rate = prm.get_double("Minimum strain rate");
         ref_strain_rate = prm.get_double("Reference strain rate");
-        min_visc = prm.get_double ("Minimum viscosity");
-        max_visc = prm.get_double ("Maximum viscosity");
         ref_visc = prm.get_double ("Reference viscosity");
-
-        AssertThrow(max_visc >= min_visc, ExcMessage("Maximum viscosity should be larger or equal to the minimum viscosity. "));
+        min_visc = Utilities::parse_map_to_double_array (prm.get("Minimum viscosity"),
+                                                         list_of_composition_names,
+                                                         has_background_field,
+                                                         "Minimum viscosity",
+                                                         true,
+                                                         expected_n_phases_per_composition);
+
+        max_visc = Utilities::parse_map_to_double_array (prm.get("Maximum viscosity"),
+                                                         list_of_composition_names,
+                                                         has_background_field,
+                                                         "Maximum viscosity",
+                                                         true,
+                                                         expected_n_phases_per_composition);
+        for (auto p1 = max_visc.begin(), p2 = min_visc.begin();
+             p1 != max_visc.end(); p1++, p2++)
```

Comment:
Not important here, but a good idea in general: (see here for more details if you are interested: https://softwareengineering.stackexchange.com/questions/358725/pre-increment-vs-post-increment)
```suggestion
             p1 != max_visc.end(); ++p1, ++p2)
```